modules/touch/touchmodule.cpp

```diff
@@ -50,68 +50,64 @@ namespace openspace {
-            [this](const Point& point) {
-        return std::find_if(
-            _listOfContactPoints.begin(),
-            _listOfContactPoints.end(),
-            [&point](const TuioCursor& c) {
-            return point.first == c.getSessionID();
-        }
-        ) == _listOfContactPoints.end(); }),
-        _lastProcessed.end());
+    _lastTouchInputs.erase(
+        std::remove_if(_lastTouchInputs.begin(), _lastTouchInputs.end(),
+                       [this](const TouchInput& input) {
+                         return std::find_if(
+                                    _touchPoints.begin(),
+                                    _touchPoints.end(),
+                                    [&input](const TouchInputs& inputs) {
+                                      return input.fingerId == inputs.getFingerId();
+                                    }) == _touchPoints.end();
+                       }),
+        _lastTouchInputs.end());
 
-    // if tap occured, we have new input
-    if (_listOfContactPoints.empty() && _lastProcessed.empty() && _ear.tap()) {
-        TuioCursor c = _ear.getTap();
-        _listOfContactPoints.push_back(c);
-        _lastProcessed.emplace_back(c.getSessionID(), c.getPath().back());
+    if(_tap) {
         _touch.tap();
+        _tap = false;
         return true;
     }
 
     // Check if we need to parse touchevent to the webgui
     processNewWebInput(_listOfContactPoints);
 
     // Return true if we got new input
-    if (_listOfContactPoints.size() == _lastProcessed.size() &&
-        !_listOfContactPoints.empty())
+    if (_touchPoints.size() == _lastTouchInputs.size() &&
+        !_touchPoints.empty())
     {
         bool newInput = true;
         // go through list and check if the last registrered time is newer than the one in
         // lastProcessed (last frame)
         std::for_each(
-            _lastProcessed.begin(),
-            _lastProcessed.end(),
-            [this, &newInput](Point& p) {
-                std::vector<TuioCursor>::iterator cursor = std::find_if(
-                    _listOfContactPoints.begin(),
-                    _listOfContactPoints.end(),
-                    [&p](const TuioCursor& c) { return c.getSessionID() == p.first; }
+            _lastTouchInputs.begin(),
+            _lastTouchInputs.end(),
+            [this, &newInput](TouchInput& input) {
+                std::vector<TouchInputs>::iterator inputs = std::find_if(
+                    _touchPoints.begin(),
+                    _touchPoints.end(),
+                    [&input](const TouchInputs& ins) {
+                        return ins.getFingerId() == input.fingerId;
+                    }
             );
-            double now = cursor->getPath().back().getTuioTime().getTotalMilliseconds();
-            if (!cursor->isMoving()) {
+            // double now = cursor->getPath().back().getTuioTime().getTotalMilliseconds();
+            if (inputs->getCurrentSpeed() == 0.0) {
```

Comment:
I'm wondering if the current speed should be compared to a very small value, or if it will be exactly zero for most cases.

Reply:
Removed in favor of the isMoving function